lang/cs/Org.Apache.REEF.Evaluator/Evaluator.cs

```diff
@@ -141,7 +142,7 @@ private static IConfiguration ReadEvaluatorConfiguration(string evaluatorConfigF
 
             AvroConfigurationSerializer serializer = new AvroConfigurationSerializer();
             var classHierarchy = TangFactory.GetTang()
-                .GetClassHierarchy(new string[] { typeof(ApplicationIdentifier).Assembly.GetName().Name });
+                .GetClassHierarchy(new string[] { typeof(ApplicationIdentifier).GetTypeInfo().Assembly.GetName().Name });
```

Comment:
I just answered my own question. The code compiles with .NET 4.5.1 as the target.